src/core/server/http/router/request.ts

```diff
@@ -141,6 +144,18 @@ export class KibanaRequest<Params = unknown, Query = unknown, Body = unknown> {
       enumerable: false,
     });
 
+    if (request.raw.req.socket instanceof TLSSocket) {
```

Comment:
I don't know if it is an expensive operation to retrieve a certificate. as we won't use if often, I'd rather avoid extra work in the constructor and declare a separate method instead
```ts
  public getPeerCertificate(detailed: boolean) {
    const request = this[requestSymbol];
    if (request.raw.req.socket instanceof TLSSocket) {
      // We request `detailed` certificate since we need to get entire certificate chain.
      const peerCertificate = request.raw.req.socket.getPeerCertificate(detailed);

      // If the peer does not provide a certificate, it returns null (if the socket has been destroyed)
      // or an empty object, so we should check for both these cases.
      if (peerCertificate && Object.keys(peerCertificate).length > 0) return peerCertificate;
    }
    return null;
  }
```

Reply:
Sure, it's up to you/core :) It's just a hacky workaround that unblocked PKI work.